nservicebus/upgrades/bridge-3to4.md

```diff
@@ -0,0 +1,29 @@
+---
+title: Messaging Bridge Upgrade Version 3 to 4
+component: Bridge
+reviewed: 2024-09-12
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 8
+ - 9
+---
+
+## ReplyToAddress header translation for failed messages
+
+Version 4 of the Messaging Bridge will translate the `NServiceBus.ReplyToAddress` message header for failed messages.
+This is a change from previous versions of the bridge, which defaulted to not translating this header.
+
+> [!NOTE]
+> The `TranslateReplyToAddressForFailedMessages()` method is now obsolete and will be removed in the next major version of the bridge.  Calls to this method should be removed
+
+Since any endpoint can be the source of a failed message, the consequence of this change in behavior is that the all endpoints need to be registered with the bridge so that the `NServiceBus.ReplyToAddress` header can be translated.
```

Comment:
```suggestion
Since the header may contain a physical address of any endpoint in the system, the change in the behavior requires all endpoints in the system to be registered with the bridge. This ensures that the information necessary for the `NServiceBus.ReplyToAddress` header translation are available when messages are transferred by the bridge.
```